cmd/kpod/bud.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/kubernetes-incubator/cri-o/libkpod"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+
+var (
+	budFlags = []cli.Flag{
+		cli.StringSliceFlag{
+			Name:  "build-arg",
+			Usage: "`argument=value` to supply to the builder",
+		},
+		cli.StringSliceFlag{
+			Name:  "file, f",
+			Usage: "`pathname or URL` of a Dockerfile",
+		},
+		cli.StringFlag{
+			Name:  "format",
+			Usage: "`format` of the built image's manifest and metadata",
+		},
+		cli.BoolTFlag{
+			Name:  "pull",
+			Usage: "pull the image if not present",
+		},
+		cli.BoolFlag{
+			Name:  "pull-always",
+			Usage: "pull the image, even if a version is present",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "refrain from announcing build instructions and image read/write progress",
+		},
+		cli.StringFlag{
+			Name:  "runtime",
+			Usage: "`path` to an alternate runtime",
+		},
+		cli.StringSliceFlag{
+			Name:  "runtime-flag",
+			Usage: "add global flags for the container runtime",
+		},
+		cli.StringFlag{
+			Name:  "signature-policy",
+			Usage: "`pathname` of signature policy file (not usually used)",
+		},
+		cli.StringSliceFlag{
+			Name:  "tag, t",
+			Usage: "`tag` to apply to the built image",
+		},
+	}
+	budDescription = "This creates an OCI image  using the 'buildah bud' command.  Buildah must be installed for this command to work."
+	budCommand     = cli.Command{
+		Name:        "bud",
+		Usage:       "Builds a container using the buildah bud command",
+		Description: budDescription,
+		Flags:       budFlags,
+		Action:      budCmd,
+		ArgsUsage:   "CONTEXT-DIRECTORY | URL",
+	}
+)
+
+func budCmd(c *cli.Context) error {
+	//args := c.Args()
+
+	config, err := getConfig(c)
+	if err != nil {
+		return errors.Wrapf(err, "Could not get config")
+	}
+	server, err := libkpod.New(config)
```

Comment:
Why do you need this at all?

Reply:
I'm not sure that I do.  I started this by grabbing the kpod ps code.  I left this in for now and suspect I'll remove it before final review.